### PR TITLE
fix get remote DTFile path

### PR DESCRIPTION
**Pull request by lidezhu**
### What problem does this PR solve?

Issue Number: ref #6827 

Problem Summary: Fix the way to get DTFile path.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/Segment.cpp [hunk 1]
```diff
@@ -1387,7 +1388,19 @@ Segment::prepareSplitLogical( //
...
-        auto file_parent_path = delegate.getDTFilePath(file_id);
+        String file_parent_path;
+        if (dm_context.db_context.getSharedContextDisagg()->remote_data_store)
+        {
+            auto wn_ps = dm_context.db_context.getWriteNodePageStorage();
+            auto full_page_id = UniversalPageIdFormat::toFullPageId(UniversalPageIdFormat::toFullPrefix(StorageType::Data, dm_context.storage_pool->getNamespaceId()), ori_page_id);
+            auto remote_data_location = wn_ps->getCheckpointLocation(full_page_id);
+            const auto & lock_key_view = S3::S3FilenameView::fromKey(*(remote_data_location->data_file_id));
+            file_parent_path = S3::S3Filename::fromTableID(lock_key_view.store_id, dm_context.storage_pool->getNamespaceId()).toFullKeyWithPrefix();
+        }
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/Segment.cpp:1399]: Can we get the parent_path from the existing `dmfile` object directly?

Comment (JaySon-Huang) [reply dbms/src/Storages/DeltaMerge/Segment.cpp:1399]: https://github.com/pingcap/tiflash/blob/4a204ea07139a1b2cff2028f1190c923e9acd8b0/dbms/src/Storages/DeltaMerge/File/DMFile.h#L458-L463

File: dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp [hunk 2]
```diff
@@ -129,7 +131,19 @@ std::vector<ColumnFilePtrT> CloneColumnFilesHelper<ColumnFilePtrT>::clone(
...
+                auto wn_ps = context.db_context.getWriteNodePageStorage();
+                auto full_page_id = UniversalPageIdFormat::toFullPageId(UniversalPageIdFormat::toFullPrefix(StorageType::Data, context.storage_pool->getNamespaceId()), f->getDataPageId());
+                auto remote_data_location = wn_ps->getCheckpointLocation(full_page_id);
+                const auto & lock_key_view = S3::S3FilenameView::fromKey(*(remote_data_location->data_file_id));
+                file_parent_path = S3::S3Filename::fromTableID(lock_key_view.store_id, context.storage_pool->getNamespaceId()).toFullKeyWithPrefix();
+            }
+            else
+            {
+                file_parent_path = delegator.getDTFilePath(file_id);
+            }
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:146]: ditto

File: dbms/src/Storages/DeltaMerge/Segment.cpp (hunk 1 above)
Comment (lidezhu) [reply dbms/src/Storages/DeltaMerge/Segment.cpp]: Fixed.

File: dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp (hunk 2 above)
Comment (lidezhu) [reply dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp]: Fixed.

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Pull request merged**